### PR TITLE
Add options needed for GDEM029T94 initialization

### PR DESCRIPTION
**Pull request by korneliuszo**
GDEM029T94 is used in 2.9inch_e-Paper_Module_V2.

### CODE REVIEW

File: drivers/display/ssd16xx.c [hunk 1]
```diff
@@ -833,6 +845,18 @@ static int ssd16xx_controller_init(const struct device *dev)
 		return err;
 	}
 
+	if (config->display_option_data.len) {
+		err = ssd16xx_write_cmd(dev, 0x37,
```
Comment (galak) [inline drivers/display/ssd16xx.c:849]: SSD16XX_CMD_OTP_SELECTION_CTRL instead of 0x37??

File: dts/bindings/display/solomon,ssd16xxfb.yaml [hunk 2]
```diff
@@ -106,3 +106,18 @@ properties:
...
+
+    data-mode-2:
+      type: boolean
+      required: false
+      description: send default updates as data2 mode
+
+    display-option-data:
+      type: uint8-array
+      required: false
+      description: setup 0x37 register for setting ping pong update
```
Comment (galak) [inline dts/bindings/display/solomon,ssd16xxfb.yaml:123]: what is the register name of 0x37.  Use that instead.

**Review by galak — CHANGES_REQUESTED** [review]
Replace magic number 0x37 in code and dts binding with register name.  Not sure if it is SSD16XX_CMD_OTP_SELECTION_CTRL

**Comment by korneliuszo** [comment]
Made second try to merge code in #49377

**Pull request closed without merging**